cirq-google/cirq_google/serialization/circuit_serializer_test.py

```diff
@@ -1204,6 +1204,16 @@ def test_custom_tag_serializer_with_tags_outside_constants():
     assert serializer.deserialize(circuit_proto) == expected_circuit
 
 
+def test_moments_with_tags():
+    serializer = cg.CircuitSerializer()
+    original_circuit = cirq.Circuit(
+        cirq.Moment(cirq.X(cirq.GridQubit(1, 1))).with_tags(DiscountTag(0.50)),
+        cirq.Moment(cirq.Z(cirq.GridQubit(2, 2))).with_tags(cg.CalibrationTag("abc")),
+    )
+    deserialized_circuit = serializer.deserialize(serializer.serialize(original_circuit))
+    assert original_circuit == deserialized_circuit
```

Comment:
Somehow the DiscountTag does not appear in `deserialized_circuit[0]`.  Please add an explicit checks for the expected moment tags.  [`Moment.__eq__`](https://github.com/quantumlib/Cirq/blob/59633f05c8be2ea2160cc436e78d5ca50db60e84/cirq-core/cirq/circuits/moment.py#L390-L394) seems to compare only the operations.

Reply:
Oh, that's a very good call!  Nice reviewing!  
I added the checks.  I forgot that the DiscountTag needs a custom serializer.  Fixed.